crypto/evp/m_sigver.c

```diff
@@ -176,6 +176,9 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
         }
 
         if (mdname != NULL) {
+            if (ctx->digest != NULL && ctx->digest->ctx_size > 0)
```

Comment:
We had fairly well separated blocks for the legacy/provider parts in this function..
Maybe this needs a comment above it.. 

What should happen here if the fetch was to fail?